frontend/src/views/expense_claim/Form.vue

```diff
@@ -409,4 +426,46 @@ function validateForm() {
 		expense.cost_center = expenseClaim.value.cost_center
 	})
 }
+
+function downloadPDF() {
+	const expenseClaimName = expenseClaim.value.name
+	loading.value = true
+
+	let headers = { "X-Frappe-Site-Name": window.location.hostname }
+	if (window.csrf_token) {
+		headers["X-Frappe-CSRF-Token"] = window.csrf_token
+	}
+
+	fetch("/api/method/hrms.api.download_expense_claim", {
+		method: "POST",
+		headers,
+		body: new URLSearchParams({ name: expenseClaimName }),
+		responseType: "blob",
+	})
+		.then((response) => {
+			if (response.ok) {
+				return response.blob()
+			} else {
+				downloadError.value = "Failed to download PDF"
+			}
+		})
+		.then((blob) => {
+			if (!blob) return
+			const blobUrl = window.URL.createObjectURL(blob)
+			const link = document.createElement("a")
+			link.href = blobUrl
+			link.download = `${expenseClaimName}.pdf`
+			link.click()
+
+			setTimeout(() => {
+				window.URL.revokeObjectURL(blobUrl)
+			}, 3000)
+		})
+		.catch((error) => {
+			downloadError.value = `Failed to download PDF: ${error.message}`
+		})
+		.finally(() => {
+			loading.value = false
+		})
+}
```

Comment:
Can we not repeat this and reuse it by moving the function out to utils?https://github.com/frappe/hrms/tree/develop/frontend/src/utils

That can be imported in Salary Slip & Expense Claim both